File: 0009_PalindromeNumber.java
class Solution {
    public boolean isPalindrome(int x) {
        if (x < 0) return false;

        int tmp = x;
        int ans = 0;
        while (x != 0) {
            ans = ans * 10 + (x % 10);
            x /= 10;
        }

        return ans == tmp;
    }
}

File: 0009_Palindrome Number.go
package main

func isPalindrome(x int) bool {
	if x < 0 {
		return false
	}

	sum, tmp := 0, x
	for tmp != 0 {
		sum = sum*10 + tmp%10
		tmp = tmp / 10
	}

	return sum == x
}
